notebooks/langchain/multi-query-retriever-examples/chatbot-with-multi-query-retriever.ipynb

```diff
@@ -0,0 +1,289 @@
+    "    json.dump(data, json_file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Split Documents into Passages\n",
+    "\n",
+    "We’ll chunk documents into passages in order to improve the retrieval specificity and to ensure that we can provide multiple passages within the context window of the final question answering prompt.\n",
+    "\n",
+    "Here we are chunking documents into 800 token passages with an overlap of 400 tokens.\n",
+    "\n",
+    "Here we are using a simple splitter but Langchain offers more advanced splitters to reduce the chance of context being lost."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.document_loaders import JSONLoader \n",
+    "from langchain.text_splitter import RecursiveCharacterTextSplitter\n",
+    "\n",
+    "def metadata_func(record: dict, metadata: dict) -> dict:\n",
+    "    metadata[\"name\"] = record.get(\"name\")\n",
+    "    metadata[\"summary\"] = record.get(\"summary\")\n",
+    "    metadata[\"url\"] = record.get(\"url\")\n",
+    "    metadata[\"category\"] = record.get(\"category\")\n",
+    "    metadata[\"updated_at\"] = record.get(\"updated_at\")\n",
+    "\n",
+    "    return metadata\n",
+    "\n",
+    "# For more loaders https://python.langchain.com/docs/modules/data_connection/document_loaders/\n",
+    "# And 3rd party loaders https://python.langchain.com/docs/modules/data_connection/document_loaders/#third-party-loaders\n",
+    "loader = JSONLoader(\n",
+    "    file_path=\"temp.json\",\n",
+    "    jq_schema=\".[]\",\n",
+    "    content_key=\"content\",\n",
+    "    metadata_func=metadata_func,\n",
+    ")\n",
+    "\n",
+    "text_splitter = RecursiveCharacterTextSplitter.from_tiktoken_encoder(chunk_size=512, chunk_overlap=256)\n",
+    "docs = loader.load_and_split(text_splitter=text_splitter)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Bulk Import Passages\n",
+    "\n",
+    "Now that we have split each document into the chunk size of 800, we will now index data to elasticsearch using [ElasticsearchStore.from_documents](https://api.python.langchain.com/en/latest/vectorstores/langchain.vectorstores.elasticsearch.ElasticsearchStore.html#langchain.vectorstores.elasticsearch.ElasticsearchStore.from_documents).\n",
+    "\n",
+    "We will use Cloud ID, Password and Index name values set in the `Create cloud deployment` step."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "documents = vectorstore.from_documents(\n",
+    "    docs, \n",
+    "    embeddings,  \n",
+    "    index_name=\"elasticsearch-multi-query-demo\",\n",
+    "    es_cloud_id=ELASTIC_CLOUD_ID,\n",
+    "    es_api_key=ELASTIC_API_KEY\n",
+    ")\n",
+    "\n",
+    "llm = OpenAI(temperature=0, openai_api_key=OPENAI_API_KEY)\n",
```

Comment:
Another deprecation warning:

    /usr/local/lib/python3.10/dist-packages/langchain_core/_api/deprecation.py:117: LangChainDeprecationWarning: The class `langchain_community.llms.openai.OpenAI` was deprecated in langchain-community 0.1.0 and will be removed in 0.2.0. Use langchain_openai.OpenAI instead.